### PR TITLE
Added MQTT example

### PR DESCRIPTION
**Pull request by HelenaMeyer**

### CODE REVIEW

File: network/mqtt_publish_subscribe.toit [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import encoding.json
+import device
+
+// Unique MQTT client ID to identify each client that connects to the MQTT broker.
+CLIENT_ID ::= "$device.hardware_id"
+// The publicly available Mosquitto MQTT server/broker is used in this example.
+HOST      ::= "test.mosquitto.org"
+// MQTT port 1883 is for unencrypted communication.
+PORT      ::= 1883
+TOPIC     ::= "my/sensor/data"
```
Comment (andersjohnsen) [inline network/mqtt_publish_subscribe.toit:17]: Add comment for this as well

File: network/mqtt_publish_subscribe.toit [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+  payload := 42.0
+
+  // Publish message to topic
+  client.publish
+    TOPIC 
+    json.encode {
+      "value": payload
+    }
+  print "Published message $payload on '$TOPIC'"
```
Comment (andersjohnsen) [inline network/mqtt_publish_subscribe.toit:40]: Move this to a function

File: network/mqtt_publish_subscribe.toit [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  // Process subscribed messages.
+  client.handle: | topic/string payload/ByteArray |
+    decoded := json.decode payload
+    print "Received message on '$topic': $(decoded["value"])"
+    // Stop after first message.
+    return
+
+subscribe client/mqtt.Client:
+  client.subscribe TOPIC --qos=1
+  print "Subscribed to MQTT client"
```
Comment (andersjohnsen) [inline network/mqtt_publish_subscribe.toit:51]: "Subscribed to topic '$TOPIC'"

File: network/mqtt_publish_subscribe.toit (hunk 2 above)
Comment (andersjohnsen) [reply network/mqtt_publish_subscribe.toit:40]: '$payload'

**Review by andersjohnsen — APPROVED** [review]

File: network/mqtt_publish_subscribe.toit (hunk 1 above)
Comment (HelenaMeyer) [reply network/mqtt_publish_subscribe.toit:17]: good point

**Pull request merged**